src/WindowsStateTriggers/EqualsStateTrigger.cs

```diff
@@ -70,27 +70,39 @@ internal static bool AreValuesEqual(object value1, object value2, bool convertTy
 			{
 				return true;
 			}
-			if (value1 != null && value2 != null)
+			if (value1 != null && value2 != null && convertType)
 			{
-				//Let's see if we can convert - for perf reasons though, try and use the right type in and out
-				if (value1.GetType() != value2.GetType() && convertType)
+				if (ConvertTypeEquals(value1, value2))
 				{
-					var t2 = System.Convert.ChangeType(value1, value2.GetType(), CultureInfo.InvariantCulture);
-					if (value2.Equals(t2))
-					{
-						return true;
-					}
-					//try the other way around
-					t2 = System.Convert.ChangeType(value2, value1.GetType(), CultureInfo.InvariantCulture);
-					if (value1.Equals(t2))
-					{
-						return true;
-					}
+					return true;
+				}
+				// Try the other way around:
+				if (ConvertTypeEquals(value2, value1))
+				{
+					return true;
 				}
 			}
 			return false;
 		}
 
+		private static bool ConvertTypeEquals(object value1, object value2)
+		{
+			// Let's see if we can convert:
+			if (value2 is Enum)
+			{
+                // Don't let Enum.ToObject() throw an exception:
+			    if (!(value1 is int) && !(value1 is Enum))
```

Comment:
It says here: https://msdn.microsoft.com/en-us/library/k6x6b517(v=vs.110).aspx
"ArgumentException when value is not type SByte, Int16, Int32, Int64, Byte, UInt16, UInt32, or UInt64."
But to be honest I am not entirely sure if it's the right way to prevent that, since int only means Int32. I don't know if it's better to write a function to check for all these possibilities or simply try{}catch{} the Enum.ToObject() call and return false in the catch. What do you think?